web/src/rtl.ts

```diff
@@ -21,26 +21,33 @@ import _ from "lodash";
  * @param {string} raw
  * @returns {number[]}
  */
-function convert_from_raw(digits, part_length, raw) {
-    const result = [];
+function convert_from_raw(digits: string, part_length: number, raw: string): number[] {
+    const result: number[] = [];
+
     for (let i = 0; i < raw.length; ) {
         let t = 0;
+
         for (let j = 0; j < part_length; j += 1) {
             t = t * digits.length + digits.indexOf(raw.charAt(i));
             i += 1;
         }
+
         result.push(t);
     }
+
     return result;
 }
 
 /** Isolate initiator characters. */
 const i_chars = new Set([0x2066, 0x2067, 0x2068]);
+
 /** Pop directional isolate character. */
 const pdi_chars = new Set([0x2069]);
+
 /** The digits that are used for base conversions from base 92. */
 const digits =
-    '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&()*+,-./:;<=>?@[]^_`{|}~';
+    '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&()*+,-./:;<=>?@[]^_`{|}~' as const;
```

Comment:
@sbansal1999 It's done to annotate its type as a constant string. So, when `as const` is used its type becomes something like this:-
```ts
const digits: "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&()*+,-./:;<=>?@[]^_`{|}~" = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&()*+,-./:;<=>?@[]^_`{|}~";
```
When people are using this constant in their code, they won't just see a generic `string` instead they'll see exactly the constant contained in that string, since this constant is never changing.

Reply:
You may not need the `as const` assertion here, as TypeScript can recognize that the assigned string is immutable. Hovering over `digits` will reveal the string's content, not just its type.

![image](https://user-images.githubusercontent.com/35286603/228866072-284a2d1e-e46d-4bdf-bd76-7196ae677308.png)